modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -108,24 +120,32 @@ public void onTopologyLeap(LogicalTopologySnapshot newTopology) {
         }
     };
 
+    /** The logical topology on the last watch event. */
+    private volatile Set<String> logicalTopology;
```

Comment:
Please add comment that will explain why volatile is enough here.

Reply:
fixed